gr-soapy/lib/block_impl.cc

```diff
@@ -1193,18 +1298,292 @@ void block_impl::cmd_handler_bw(pmt::pmt_t val, size_t channel)
 {
     if (!val->is_symbol()) {
-        GR_LOG_WARN(d_logger, "soapy: ant must be string");
+        GR_LOG_ERROR(d_logger, "soapy: ant must be string");
         return;
     }
     set_antenna(channel, pmt::symbol_to_string(val));
 }
 
+void block_impl::cmd_handler_gain_mode(pmt::pmt_t val, size_t channel)
+{
+    if (!val->is_bool()) {
+        GR_LOG_ERROR(d_logger, "soapy: gain mode must be bool");
+        return;
+    }
+    set_gain_mode(channel, pmt::to_bool(val));
+}
+
+void block_impl::cmd_handler_frequency_correction(pmt::pmt_t val, size_t channel)
+{
+    if (!(val->is_number() && !val->is_complex())) {
+        GR_LOG_ERROR(d_logger, "soapy: frequency correction must be float/int");
+        return;
+    }
+    set_frequency_correction(channel, pmt::to_double(val));
+}
+
+void block_impl::cmd_handler_dc_offset_mode(pmt::pmt_t val, size_t channel)
+{
+    if (!val->is_bool()) {
+        GR_LOG_ERROR(d_logger, "soapy: DC offset mode must be bool");
+        return;
+    }
+    set_dc_offset_mode(channel, pmt::to_bool(val));
+}
+
+void block_impl::cmd_handler_dc_offset(pmt::pmt_t val, size_t channel)
+{
+    if (!val->is_number()) {
+        GR_LOG_ERROR(d_logger, "soapy: DC offset must be numeric");
+        return;
+    }
+    set_dc_offset(channel, pmt::to_complex(val));
+}
+
+void block_impl::cmd_handler_iq_balance(pmt::pmt_t val, size_t channel)
+{
+    if (!val->is_number()) {
+        GR_LOG_ERROR(d_logger, "soapy: IQ balance must be numeric");
+        return;
+    }
+    set_iq_balance(channel, pmt::to_complex(val));
+}
+
+void block_impl::cmd_handler_iq_balance_mode(pmt::pmt_t val, size_t channel)
+{
+    if (!val->is_bool()) {
+        GR_LOG_ERROR(d_logger, "soapy: IQ balance mode must be bool");
+        return;
+    }
+    set_iq_balance_mode(channel, pmt::to_bool(val));
+}
+
+void block_impl::cmd_handler_master_clock_rate(pmt::pmt_t val, size_t)
+{
+    if (!(val->is_number() && !val->is_complex())) {
+        GR_LOG_ERROR(d_logger, "soapy: master clock rate must be float/int");
+        return;
+    }
+    set_master_clock_rate(pmt::to_double(val));
+}
+
+void block_impl::cmd_handler_reference_clock_rate(pmt::pmt_t val, size_t)
+{
+    if (!(val->is_number() && !val->is_complex())) {
+        GR_LOG_ERROR(d_logger, "soapy: reference clock rate must be float/int");
+        return;
+    }
+    set_reference_clock_rate(pmt::to_double(val));
+}
+
+void block_impl::cmd_handler_clock_source(pmt::pmt_t val, size_t)
+{
+    if (!val->is_symbol()) {
+        GR_LOG_ERROR(d_logger, "soapy: clock source must be string");
+        return;
+    }
+    set_clock_source(pmt::symbol_to_string(val));
+}
+
+void block_impl::cmd_handler_time_source(pmt::pmt_t val, size_t)
+{
+    if (!val->is_symbol()) {
+        GR_LOG_ERROR(d_logger, "soapy: time source must be string");
+        return;
+    }
+    set_time_source(pmt::symbol_to_string(val));
+}
+
+void block_impl::cmd_handler_hardware_time(pmt::pmt_t val, size_t)
```

Comment:
Why a dict here vs uint64_t?

Reply:
https://github.com/gnuradio/gnuradio/blob/master/gr-soapy/include/gnuradio/soapy/block.h#L466

At the bottom, the handler calls into this function, which optionally takes in a name. I need a dict to take both the name and time.